### PR TITLE
Feature/feedback management

### PR DESCRIPTION
**Pull request by arslanyousaf77**

### CODE REVIEW

File: app/assets/stylesheets/feedback-form.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+.stars
+{
+    margin: 20px 0;
+    font-size: 26px;
+    color: #F0A419;
+}
+
+h4{
+    color:#5F6368;
+}
```
Comment (abbasraza172) [inline app/assets/stylesheets/feedback-form.css]: add new line at end.

File: app/controllers/feedbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    path = validate_feedback(feedback_params)
+    unless path.nil?
+      return redirect_to path
+    end
+    @feedback = Feedback.new(feedback_params)
+    if @feedback.save
+      @order = Order.find(feedback_params[:order_id])
+      path = give_next_feedback_path
+    else
+      # render :action => "create"
```
Comment (abbasraza172) [inline app/controllers/feedbacks_controller.rb]: why commented code ?

File: app/controllers/feedbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  #It receives params and validates them whether given params
+  #for feedback are valid or not. If they are valid, it returns nil.
+  #If not valid, it returns index_path.
+  def validate_feedback(feedback_params)
+    path = nil
+    order = Order.find(feedback_params[:order_id])
+    if order && order.user_id == feedback_params[:user_id].to_i
+      if feedback_params[:reviewable_type].camelize == "Product"
+        if !order.products.ids.include?(feedback_params[:reviewable_id].to_i)
+          path = index_path
```
Comment (abbasraza172) [inline app/controllers/feedbacks_controller.rb]: why do we have different conditions that set same value in path. ?
does it help?

File: app/models/order.rb [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
+  after_update :send_feedback_email
+
+  #Sends email to customer for feedback
+  #
+  #It checks if the status of order is changed form
+  #any status other than "delivered" to "delivered"
+  #it sends an email to order's customer to give feedback
+  def send_feedback_email
+    if self.status_before_last_save != "delivered" && self.delivered?
+      puts "Hello"
```
Comment (abbasraza172) [inline app/models/order.rb]: puts ?

File: app/views/home/index.html.erb [hunk 5]
```diff
@@ -8,6 +8,7 @@
     <%= render partial: "shared/default/productsCards", locals: {header: "New Products", products: @new_products} %>
     <%= render partial: "shared/default/productsCards", locals: {header: "Featured Products", products: @featured_products} %>
     <%= render partial: "shared/default/productsCards", locals: {header: "Popular Products", products: @popular_products} %>
+    <button id="myBtn" onclick = "window.myFunction(this) ">Try it</button>
```
Comment (abbasraza172) [inline app/views/home/index.html.erb]: ?? again try it ?

File: app/controllers/feedbacks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,94 @@
+class FeedbacksController < ApplicationController
+  layout "feedback_layout"
+
+  def new
+    path = validate_feedback(params)
+    unless path.nil?
+      return redirect_to path
+    end
+
+    # @order = Order.find(params[:order_id])
```
Comment (abbasraza172) [inline app/controllers/feedbacks_controller.rb:10]: commented code ?

File: app/javascript/packs/default_layout.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ import Rails from "@rails/ujs"
 import Turbolinks from "turbolinks"
 import * as ActiveStorage from "@rails/activestorage"
 import "channels"
-
+require("jquery")
```
Comment (abbasraza172) [inline app/javascript/packs/default_layout.js:11]: please add extra line on top

File: config/env.yml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+CONTACT: 0900 786011
```
Comment (abbasraza172) [inline config/env.yml]: this file should be removed. and not added in commit

File: config/initializers/constant.rb [hunk 9]
```diff
@@ -1 +1,2 @@
 APP_NAME = "Coeus Express"
+REDIRECT_BUYER_DOMAIN = "buyer.lvh.me:300"
```
Comment (abbasraza172) [inline config/initializers/constant.rb]: this is not required anymore right ?

File: config/env.yml (hunk 8 above)
Comment (arslanyousaf77) [reply config/env.yml]: okay

File: config/initializers/constant.rb (hunk 9 above)
Comment (arslanyousaf77) [reply config/initializers/constant.rb]: yeah but our issue is still unresolved for what we had added this.

File: app/controllers/feedbacks_controller.rb (hunk 6 above)
Comment (arslanyousaf77) [reply app/controllers/feedbacks_controller.rb:10]: This is pending feature basically. It was added to check whether a customer has visited a link for review or not. I am removing it btw for now.

File: config/initializers/constant.rb (hunk 9 above)
Comment (abbasraza172) [reply config/initializers/constant.rb]: okay

**Review by abbasraza172 — APPROVED** [review]

**Pull request merged**